modules/photo/src/inpaint.cpp

```diff
@@ -84,42 +94,10 @@ class CvPriorityQueueFloat
     CvPriorityQueueFloat& operator=(const CvPriorityQueueFloat &); // assign disabled
 
 protected:
-    CvHeapElem *mem,*empty,*head,*tail;
```

Comment:
I propose to remove `CvPriorityQueueFloat` as it is not public API and replace its usage with std object calls. cv::Ptr is not needed any more and you can use `std::priority_queue&` as parameter.

Reply:
I see. That is a good idea. I'll fix it.

Reply:
Actually I need to add `next_order` variable in the priority queue class to keep insertion order for complete compatibility. It is because `std::priority_queue` cannot keep insertion order when two elements have the same score.
So I think it is simpler to keep this class. If it is OK to break compatibility, i can remove this variable. Even in that case, the result is almost same. How do you think?